runtime/src/main/cpp/Memory.cpp

```diff
@@ -402,7 +401,8 @@ class ForeignRefManager {
 
       if (hadNoStateInitialized) {
         // Discard the memory state.
-        DeinitMemory(memoryState);
+        if (!DeinitMemory(memoryState))
+          konan::abort();
```

Comment:
What was moving `abort` out from `DeinitMemory` exactly required for?

Reply:
That used to be for consistency with `WorkerDeinit`, where I didn't use `abort` in order to run memory leak checker even after leaked workers were found.

However, current implementation won't allow running memory leak checker in that situation anyway, so I restored `abort` here and in `WorkerDeinit`.